### ISSUE TITLE
scripts for missing superuniues

### ISSUE THREAD

**Issue by amOKchen** [kolton/d2bot-with-kolbot#1665]
My MF bots are running most default superuniques from act1 to 5.

But the default kolbot scripts does not list all superuniques.
Is there anywhere you can find scripts for superuniques like bishibosh, coldcrow and etc?

**Comment by DarkHorseDre** [comment]
Boss scripts are written specifically for each boss/set of bosses.  The main ones are covered, but maybe not all.  Kolton is provided so we can extend if we want, and share if we want (many have contributed to date)

The boss files are in your repo, as: https://github.com/kolton/d2bot-with-kolbot/tree/master/d2bs/kolbot/libs/bots

You'll find coldcrow here and in your config file (just enable it)

Bishibosh is identified in attack.js but not specific boss script seems to exist, so you'd have to copy one of the boss scripts and change to him (all same sequence:  town chores, wp to area, travel to boss coords, attackkill by id)

**Comment by amOKchen** [comment]
Seems rather simple to make a killscript when i look at the ones that already exists.

However I cant locate where to find documentation for two things in pather, unitType and unitId.
It is not in kolbot\sdk\superunique_presetunitids.txt or SuperUniques.txt, at far as i Can tell.

```
function Bishibosh() {
	Town.doChores();
	Pather.useWaypoint(3);
	Precast.doPrecast(true);

	if (!Pather.moveToPreset(me.area, **?**, **?**, 5, 5)) { 
		throw new Error("Failed to move to Bishibosh");
	}

	Attack.clear(15, 0, getLocaleString(2869)); // Bishibosh

	return true;
}
```

**Comment by amOKchen** [comment]
So i figured it out, and I intend to make more scripts later on when I have time.
My bot farms bishibosh just fine now with this.

superunique_presetunitids.txt was a bit misleading though, figured out on a german site that some monsters on that list needs to add 30.
So bishibosh has id 734 and not 704 like it shows.

@DarkHorseDre where do I contribute if this is relevant?

```
function Bishibosh() {
	Town.doChores();
	Pather.useWaypoint(3);
	Precast.doPrecast(true);

	if (!Pather.moveToPreset(me.area, 1, 734, 0, 0)) { // area, unitType, unitId, offX, offY, clearPath, pop (
		throw new Error("Failed to move to Bishibosh");
	}

	Attack.clear(15, 0, getLocaleString(2869)); // Bishibosh

	return true;
}
```

**Comment by DarkHorseDre** [comment]
yeah there's something written somewhere (in the codebase?) about adding numbers to get the correct id, gj finding it lol

you can also find out stuff on mouse pointer with `print(JSON.stringify(getUnit(101).spectype));` but change the property to whatever you need

You can raise it as a PR but you need to conform to the preferred style and format conventions and I have no idea what that is

check the pull request tab

**Comment by amOKchen** [comment]
How do you go about activating that via d2bs?
Does not seem to do much via console.

**Comment by DarkHorseDre** [comment]
place the mouse pointer at a mob, ALT+HOME to toggle text entry mode, execute code in console, it will print out the object data.

check the api.html for getunit.  check out the other entries too.
I think there are other resources out there but dunno where..

**Comment by WolfOfWallStD2Jjsp** [comment]
> place the mouse pointer at a mob, ALT+HOME to toggle text entry mode, execute code in console, it will print out the object data.
> 
> check the api.html for getunit. check out the other entries too.
> I think there are other resources out there but dunno where..

How do you use this ? ... So I press alt home while I have cursor on say Bishibosh then check this file you mentioned and it gives the info I need to put in the script ? 

this is all so new

**Comment by DarkHorseDre** [comment]
I didnt mention a file - read the instructions and follow as-is

**Comment by ircopdd** [comment]
> 我沒有提及文件-請閱讀說明並按原樣進行
> 
> 這也是一個死線程，但我注意到我使用了不正確的命令-現在已更新
> （print（getUnit（101）.spectype）;）

It doesn't work

**Comment by DarkHorseDre** [comment]
print(getUnit(101).spectype);

> > 我沒有提及文件-請閱讀說明並按原樣進行
> > 這也是一個死線程，但我注意到我使用了不正確的命令-現在已更新
> > （print（getUnit（101）.spectype）;）
> 
> It doesn't work

You're doing it wrong.

enter the command with the mouse over a monster `print(getUnit(101).spectype);`

if that doesnt work, troubleshoot what you are doing.

**Comment by ircopdd** [comment]
> print(getUnit(101).spectype);
> 
> > > 我沒有提及文件-請閱讀說明並按原樣進行
> > > 這也是一個死線程，但我注意到我使用了不正確的命令-現在已更新
> > > （print（getUnit（101）.spectype）;）
> > 
> > 
> > It doesn't work
> 
> You're doing it wrong.
> 
> enter the command with the mouse over a monster `print(getUnit(101).spectype);`
> 
> if that doesnt work, troubleshoot what you are doing.

thanks 

That's the right instruction

**Comment by DarkHorseDre** [comment]
lol its the same as the previous but in brackets - no js starts with brackets ofc
